### PR TITLE
Add Moonshine speech recognition support

### PR DESCRIPTION
**Pull request by justinchuby**
## Summary

- add native `moonshine` support for `moonshine-ai/moonshine-tiny` without aliasing it to Whisper
- implement the raw-waveform convolution encoder, GroupNorm, partial interleaved RoPE, bias-free encoder/decoder attention, gated decoder MLP, cross-attention masking, tied output projection, and cached decoding
- generalize `SpeechToTextTask` while preserving Whisper, including correct 80/128 `num_mel_bins` input-channel handling
- export tokenizer, `audio_processor.json`, and composite onnx-genai metadata
- add L1 graph/config/alignment, L2 real-config, L3 synthetic/real-weight parity, and real-audio L4/L5 goldens

## Updated from main

Rebased onto `origin/main` at `1e6f61f42b0ddbc90f60bccbf00eff26cd484256`, including main's fused Swish support. Moonshine now uses the public `SiLU` component and emits one ONNX `Swish` op per decoder layer instead of `Sigmoid + Mul`.

## Fusion and optimized-graph audit

Audited fp16 opset-24 graphs with ONNX Runtime GPU 1.28.0 on CUDA EP; `CUDAExecutionProvider` was asserted first for encoder and decoder.

| Graph | Mobius nodes | ORT-optimized nodes | Key optimized ops | Identity | Unused initializers | Memcpy |
|---|---:|---:|---|---:|---:|---:|
| encoder | 115 | 131 | 6 Attention, 6 BiasGelu, 12 SkipLayerNormalization | 0 | 0 | 0 |
| decoder | 151 | 151 | 12 Attention, 6 QuickGelu(alpha=1.0), 18 SkipLayerNormalization | 0 | 0 | 0 |

Additional structural evidence:

- Mobius decoder graph: 6 `Swish`, 0 `Sigmoid`
- ORT folds each Swish to `QuickGelu(alpha=1.0)`
- encoder GroupNormalization is optimized without CPU fallback
- only one encoder transpose remains, required for Conv1d `[B,C,T]` to transformer `[B,T,C]`
- no protobuf construction APIs
- no opset-lowering workaround needed on ORT 1.28

The standard ONNX `Attention` op is intentional: encoder and cross-attention require explicit padding bias, which contrib GQA rejects, and Moonshine's head dimension 36 is not XQA-eligible.

### CPU and CUDA parity

Environment: NVIDIA RTX A1000, driver 573.44, CUDA 12.8, ONNX Runtime GPU 1.28.0.

Real `652-129742-0006.flac`, fp32:

| Boundary | Max abs diff | Mean abs diff | Result |
|---|---:|---:|---|
| encoder hidden states | 0.0045447 | 0.0001437 | finite; masks exact |
| decoder prefill logits | 0.0098586 | 0.0011758 | finite; identical argmax 29892 |
| greedy transcription | exact | exact | CPU == CUDA == 32-token golden |

### Dtype coverage

- fp16 CUDA: finite encoder/decoder state and exact 32/32 fp32-golden transcription tokens
- bf16 CUDA: finite state and exact Hugging Face bf16 CUDA generation tokens; bf16 output differs from fp32 at the expected near-tie prefix

### Test commands/results

- `python -m pytest tests/build_graph_test.py -k "moonshine or whisper" -q`: 17 passed
- `python -m pytest tests/moonshine_integration_test.py -q`: 5 passed
- speech config/alignment/parity selection: 18 passed
- CUDA `tests/e2e_golden_test.py -m "golden or generation" -k moonshine`: 2 passed
- fusion/task/export selection: 13 passed
- YAML schema + model coverage: 945 passed, 215 skipped
- `lintrunner -a`: no lint issues
- automated review findings resolved, including explicit Whisper 128-mel config/graph regression coverage

## Checklist / waivers

- [x] L1 graph construction and weight alignment
- [x] L2 real Hugging Face config build
- [x] L3 synthetic and real-weight encoder/decoder/cache parity
- [x] L4 real-audio logits golden on CPU and CUDA
- [x] L5 deterministic transcription golden on CPU and CUDA
- [x] fp32/fp16/bf16 CUDA validation
- [x] CLI export, processor/tokenizer wiring, runtime load, and generation
- [x] ORT 1.28 optimized-graph and fusion audit with zero Memcpy
- [x] full `lintrunner -a`, schema, coverage, and targeted regressions

**Waivers:**

- **Native onnxruntime-genai:** upstream's audio-language runtime is Whisper-specific and hardcodes 3000 input / 1500 encoder frames plus Whisper alignment tensors. Moonshine uses variable-length raw waveforms, so this PR explicitly rejects falsely mapping it to Whisper and validates onnx-genai metadata plus direct ONNX Runtime CUDA generation instead.
- **Foundry Local:** not installed in this environment.
- **Olive quantization:** Olive is not installed in this environment.

### CODE REVIEW

File: src/mobius/integrations/ort_genai/auto_export.py [hunk 1]
```diff
@@ -1008,6 +1008,13 @@ def write_ort_genai_config(
             "Diffusion models (which have no config) are not supported."
         )
 
+    if getattr(config, "model_type", None) == "moonshine":
+        raise NotImplementedError(
+            "onnxruntime-genai does not support Moonshine's variable-length raw-waveform "
+            "encoder. Export with runtime='onnx-genai' or run the encoder and cached "
```
Comment (justinchuby) [inline src/mobius/integrations/ort_genai/auto_export.py]: onnx-genai is experimental. Avoid mentioning in this path

File: src/mobius/models/moonshine.py [hunk 2]
```diff
@@ -0,0 +1,466 @@
...
+        op.Shape(attention_mask, start=1, end=2),
+        axis=0,
+    )
+    valid = op.Expand(valid, target_shape)
+    bias = op.Where(valid, 0.0, float(dtype.min))
+    return op.Cast(bias, to=dtype)
+
+
+class MoonshineGroupNorm(nn.Module):
+    """Single-group normalization without the CPU-only ONNX GroupNormalization op."""
```
Comment (justinchuby) [inline src/mobius/models/moonshine.py]: We should use onnx group norm

File: testdata/cases/audio/moonshine-tiny.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+model_id: "moonshine-ai/moonshine-tiny"
+model_type: "moonshine"
+revision: "390624ed33d594443aa4aa221f5b9f283b545b5a"
+task_type: "speech-to-text"
+dtype: "float32"
+
+inputs:
+  audio:
+    - "652-129742-0006.flac"
```
Comment (justinchuby) [inline testdata/cases/audio/moonshine-tiny.yaml:9]: Is there L5 json generated?

**Review by justinchuby — COMMENTED** [review]
Reviewed the current diff for correctness; no additional high-confidence issues found beyond the existing unresolved review threads.

File: src/mobius/tasks/_speech_to_text.py [hunk 4]
```diff
@@ -25,7 +25,7 @@
 
 
 class SpeechToTextTask(ModelTask):
-    """Encoder-decoder speech-to-text task (e.g. Whisper).
+    """Encoder-decoder speech-to-text task.
 
     This task builds **two** separate ONNX models via :meth:`build`:
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/tasks/_speech_to_text.py]: Fixed in aeb1446. SpeechToTextTask documentation now describes the configured encoder input name, optional encoder mask, conditional decoder mask routing, and the concrete Whisper/Moonshine contracts; encoder/decoder helper docstrings were updated too.

File: src/mobius/integrations/ort_genai/auto_export.py (hunk 1 above)
Comment (justinchuby) [reply src/mobius/integrations/ort_genai/auto_export.py]: Fixed in aeb1446. The ORT GenAI rejection no longer suggests the experimental onnx-genai path; it directs users to the exported encoder and cached decoder through ONNX Runtime.

File: src/mobius/models/moonshine.py (hunk 2 above)
Comment (justinchuby) [reply src/mobius/models/moonshine.py]: Fixed in aeb1446. Removed the custom reduction implementation and now use the public ONNX `GroupNorm(1, hidden_size)` component with the HF epsilon. Targeted numerical parity and graph tests pass.

File: testdata/cases/audio/moonshine-tiny.yaml (hunk 3 above)
Comment (justinchuby) [reply testdata/cases/audio/moonshine-tiny.yaml:9]: Confirmed in aeb1446. The genuine deterministic L5 generation JSON already existed and regeneration matched it exactly. Added a test that discovers the Moonshine L5 case, requires the generation file, loads it, and validates non-empty integer token IDs.

File: tests/moonshine_integration_test.py [hunk 5]
```diff
@@ -0,0 +1,357 @@
+# Copyright (c) Microsoft Corporation.
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply tests/moonshine_integration_test.py:1]: Fixed in 922628e by applying the repository formatter to the Moonshine L5 reference test.
